scripts/bls/eigen/register.sh

```diff
@@ -38,6 +48,21 @@ ensure_balance "$OP_ADDR"
 echo "Operator address: $OP_ADDR"
 echo "Delegate amount: $WAVS_DELEGATE_AMOUNT"
 
-# Register operator
 cd contracts || handle_error "Failed to change to contracts directory"
-forge script script/eigenlayer/bls/WavsRegisterOperator.s.sol -vvv --rpc-url "$RPC_URL" --private-key "$OPERATOR_KEY" --broadcast --skip-simulation --slow || handle_error "Failed to register operator"
+
+# Deposit into strategy
+forge script script/eigenlayer/bls/WavsDepositIntoStrategy.s.sol -vvv --rpc-url "$RPC_URL" --private-key "$OPERATOR_KEY" --broadcast || handle_error "Failed to deposit into strategy"
+
+# Wait based on environment
+if [[ "$DEPLOY_ENV" == "LOCAL" ]]; then
+    echo "Mining $ALLOCATION_DELAY blocks for local transaction confirmation..."
+    for ((i=1; i<=ALLOCATION_DELAY; i++)); do 
+        cast rpc evm_mine --rpc-url http://localhost:8545
+    done
+elif [[ "$DEPLOY_ENV" == "TESTNET" ]]; then
+    WAIT_TIME=$((12 * ALLOCATION_DELAY))
```

Comment:
This seems a bit arbitrary.

Could you do something like:

* calculate target block number (current + allocation delay)
* while current_block < target_block
  * sleep (5 * target - current)

This is more robust and doesn't have to wait 2x the expected time. You can change 5 with any other number you like, whatever would be a lower bound for block time (could even be 1, you just have a few extra queries, but that is cheap)